### PR TITLE
Convert drivers from imports to strings

### PR DESCRIPTION
**Pull request by Emantor**
**Description**
This PR fixes the lookup by string during the driver binding by introducing a new dictionary in the TargetFactory and using this to look at all registered resources and drivers.
It then converts the imports used for the bindings to strings.

**Checklist**
- [ ] Documentation for the feature
- [ ] Tests for the feature 
- [x] PR has been tested

Tested by running tox, which also revealed the lookup problem.

### CODE REVIEW

File: labgrid/driver/serialdriver.py [hunk 1]
```diff
@@ -38,7 +37,7 @@ class SerialDriver(ConsoleExpectMixin, Driver, ConsoleProtocol):
     def __attrs_post_init__(self):
         super().__attrs_post_init__()
         self.logger = logging.getLogger("{}({})".format(self, self.target))
-        if isinstance(self.port, SerialPort):
+        if isinstance(self.port, target_factory.registered_class_from_string("SerialPort")):
```
Comment (jluebbe) [inline labgrid/driver/serialdriver.py]: Please don't replace static instances in normal code (for readability and performance reasons).

File: labgrid/factory.py [hunk 2]
```diff
@@ -147,6 +152,18 @@ def make_target(self, name, config, *, env=None):
             self.make_driver(target, driver, name, args)
         return target
 
+    def registered_class_from_string(self, string: str):
+        print(self.all_classes)
```
Comment (jluebbe) [inline labgrid/factory.py]: Leftover debug print.

File: labgrid/factory.py [hunk 3]
```diff
@@ -1,23 +1,28 @@
+import inspect
```
Comment (jluebbe) [inline labgrid/factory.py]: s/not bound/not registered/ in the commit message ("bound" has a different meaning in labgrid).

File: labgrid/target.py [hunk 4]
```diff
@@ -110,7 +110,7 @@ def get_resource(self, cls, *, name=None, wait_avail=True):
         found = []
         other_names = []
         if isinstance(cls, str):
-            cls = self._class_from_string(cls)
+            cls = self._bound_class_from_string(cls)
```
Comment (jluebbe) [inline labgrid/target.py]: Why not registered?

File: labgrid/factory.py [hunk 5]
```diff
@@ -147,6 +152,18 @@ def make_target(self, name, config, *, env=None):
             self.make_driver(target, driver, name, args)
         return target
 
+    def registered_class_from_string(self, string: str):
```
Comment (jluebbe) [inline labgrid/factory.py]: The name is wrong (and too long to be convinient). Currently, this function will find even unregistered parent classes.

**Comment by Emantor** [comment]
Updated. Please take a look again.

**Comment by Emantor** [comment]
> "bound" is wrong here. I'd suggest:
> 
> > With this we can lookup classes which have been registered directly, but their base classes as well.

And I thought I got all cases. Fixed.

**Pull request merged**